mne/epochs.py

```diff
@@ -2628,9 +2628,16 @@ def add_annotations_to_metadata(self):
         # onsets, durations, and descriptions
         epoch_annot_list = self.get_annotations_per_epoch()
         onset, duration, description = [], [], []
-        for epoch in epoch_annot_list:
+        for epoch_annot in epoch_annot_list:
             for ix, annot_prop in enumerate((onset, duration, description)):
-                entry = [annot[ix] for annot in epoch] if len(epoch) else []
+                entry = [annot[ix] for annot in epoch_annot]
+
+                # round onset and duration to avoid IO round trip mismatch
+                if ix < 2 and len(entry):
+                    entry = list(
+                        map(lambda x: float(np.round(x, decimals=12)), entry)
+                    )
```

Comment:
Why not just
```suggestion
                    entry = np.round(x, decimals=12).tolist()
```
? And the `tolist` maybe isn't even necessary.?

Reply:
(if you need it, you could add a `.astype(float)` above, too -- to me it's just nice to avoid both a Python iteration over all elements and a `map` call)

Reply:
better yet, why not apply the rounding step below when `onset` and `duration` are converted to `pd.Series`?  it moves it out of the for-loop, eliminates the `if ix < 2` conditional, and should be possible without a `map` or `lambda`.

Reply:
onset and duration are list of list so  the pd.Series is of dtype object. I cannot do it as this stage I think @drammock

Reply:
ah, right, I forgot the series were dtype object